src/plugins/data_source_management/public/management_app/mount_management_section.tsx

```diff
@@ -48,17 +48,21 @@ export async function mountManagementSection(
     authenticationMethodRegistry: authMethodsRegistry,
   };
 
+  const canManageDataSource = !!application.capabilities?.dataSource?.canManage;
+
   ReactDOM.render(
     <OpenSearchDashboardsContextProvider services={deps}>
       <I18nProvider>
         <Router history={params.history}>
           <Switch>
             <Route path={['/create']}>
-              <CreateDataSourcePanel {...params} featureFlagStatus={featureFlagStatus} />
+              {canManageDataSource ? (
+                <CreateDataSourcePanel {...params} featureFlagStatus={featureFlagStatus} />
+              ) : null}
             </Route>
             {featureFlagStatus && (
               <Route path={['/configure/OpenSearch']}>
-                <CreateDataSourceWizardWithRouter />
+                {canManageDataSource ? <CreateDataSourceWizardWithRouter /> : null}
               </Route>
             )}
             <Route path={['/configure/:type']}>
```

Comment:
I think we need to also include this route under the flag, so that the 
`http://localhost:5601/app/management/opensearch-dashboards/dataSources/configure/Prometheus` and `http://localhost:5601/app/management/opensearch-dashboards/dataSources/configure/AmazonS3AWSGlue` Will also be filtered out.

Reply:
added